### DISCUSSION TITLE
`Notification API`s discussion.subject has no url

### DISCUSSION THREAD

**Discussion by zunda-pixel** [community/community#62728] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

I want to get discussion number like `#123`

[List notifications for the authenticated user](https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#list-notifications-for-the-authenticated-user)'s discussion.subject has no url

issue: OK
release: OK
pullRequest: OK
discussion: NO

I know REST API has no discussion endpoint.
Please add `number field`, or normal url https://github.com/orgs/community/discussions/62728

**Comment by leeoniya** [comment]
just ran into this as well, any updates?

**Reply by zunda-pixel** [reply]
nothing for now

**Comment by LucasCoderT** [comment]
I'm currently blocked by this as well. 

However I've seen an app be able to get the discussion from the notification just fine so it might be possible, but this app is closed source so I cannot find out how they do it.

**Comment by wottpal** [comment]
Same issue, did anyone find a workaround? For me it seems as there is currently no way to get a discussion number for a given notification thread.

**Reply by LangLangBart** [reply]
> Same issue, did anyone find a workaround?

A **workaround** is to gather information from calling the `Notification REST API` and then use it in the `GraphQL API` to make a search with specific parts of this information.

---

In the **example** below, the following information is collected from the `Notification REST API`:
- `.subject.title`
- `.updated_at`
- `.repository.full_name`

These pieces of information are parsed into a string for a `GraphQL` search query[^1]. Search **qualifiers**, as described in the docs[^2], are used to make the search more specific. 


#### Query
```bash
graphql_query_discussion=$'query ($filter: String!) {
  search(query: $filter, type: DISCUSSION, first: 1) {
    nodes {
      ... on Discussion { number url }
    }
  }
}'

# [gh] https://github.com/cli/cli
gh api notifications \
    --cache=1m \
    --method GET \
    --field all=true \
    --field per_page=10 \
    --jq '.[] | select(.subject.type == "Discussion") |
        "repo:\(.repository.full_name) updated:>=\(.updated_at | fromdateiso8601 | strftime("%Y-%m")) in:title \(.subject.title)"' |
    while IFS=$'\n' read -r line; do
        echo "[INPUT] $line"
        gh api graphql \
            --cache=1h \
            --raw-field filter="$line" \
            --raw-field query="$graphql_query_discussion" \
            --jq '.data.search.nodes[] | "[RESULT] Number: \(.number)\tURL: \(.url)"'
        echo
    done
```

#### Result
```txt
[INPUT] repo:community/community updated:>=2024-04 in:title Getting user commit count for the month
  [RESULT] Number: 114470  URL: https://github.com/orgs/community/discussions/114470
```

>[!NOTE]
> This is a hacky solution. While it increases the likelihood of locating the desired data, it doesn't
> guarantee results.
>
> A more effective solution would be for Github to implement the OP's suggestion of including the
> number and URL in the `Notification REST API` response. Alternatively, Github could enable
> notification queries in the `GraphQL API`[^3]. At present, users must navigate two APIs, and 
> obtaining the correct data still remains uncertain.

[^1]: [Search Queries - GitHub Docs GraphQL](https://docs.github.com/en/graphql/reference/queries#search)
[^2]: [Searching discussions - GitHub Docs](https://docs.github.com/en/search-github/searching-on-github/searching-discussions)
[^3]: [Notifications GraphQL API · community · Discussion #13056 · GitHub](https://github.com/orgs/community/discussions/13056)

**Reply by wottpal** [reply]
Never got such an extensive and well-written reply before haha. Work just fine! ❤️

Still, GitHub should just return the discussion number and/or url in the first place.

**Comment by zunda-pixel** [comment]


**Reply by queenofcorgis** [reply]
Hi there,

Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our Code of Conduct for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

**Reply by zunda-pixel** [reply]
Sorry, I didn't know.

**Comment by JackCoplandGranta** [comment]
The same seems to be true for notifications with reason "approval_requested", the url is just blank, and there is no way to know build the notification originated from.
